### PR TITLE
Skip ClearMempool when node is leader in upcoming round

### PR DESCRIPTION
**Pull request by andr-dev**
If the current node will be the leader in the next few rounds, we should not clear the mempool.

### CODE REVIEW

**Comment by omegablitz** [comment]
I think the single node devnet txpool will grow unboundedly if ingress rate exceeds block production rate

**Comment by ariqchowdhury** [comment]
> I think the single node devnet txpool will grow unboundedly if ingress rate exceeds block production rate

That should be fine on `devnet-prime`
But we can also add a hard-cap to the txpool size in case -- just clear everything no matter what after a proposal if there's X number of txns

**Comment by michael-yxchen** [comment]
I think it's helpful to have a hard cap. Reasoning about leader schedule and `NUM_LEADERS_FORWARD_SKIP_MEMPOOL_CLEAR` can get confusing. i.e. 3 equal-stake nodes with MEMPOOL_CLEAR configured as 3 will never clear their mempool

**Comment by andr-dev** [comment]
New idea from @ariqchowdhury is to tell the txpool in `create_proposal` whether we'll also be an upcoming validator so it can decide whether to move the entire pool to the `garbage`, effectively clearing itself, or to just remove the txs that get proposed. I pushed up the changes needed outside of the txpool, now working through the txpool itself.

File: monad-consensus-state/src/lib.rs [hunk 1]
```diff
@@ -1205,13 +1208,33 @@ where
...
+                    })
+                    .any(|(epoch, round)| {
+                        &self.election.get_leader(
+                            round,
+                            self.val_epoch_map
+                                .get_val_set(&epoch)
+                                .expect("handle non-existent validatorset for upcoming round epoch")
+                                .get_members(),
+                        ) == self.nodeid
+                    });
```
Comment (andr-dev) [inline monad-consensus-state/src/lib.rs:1228]: Will become a helper function in a bit.

**Comment by ariqchowdhury** [comment]
new txpool design doesn't get rid of anything (unless at capacity) so this PR can be closed

**Pull request closed without merging**